shopping-cart-impl/src/main/java/com/example/shoppingcart/impl/ShoppingCartException.java

```diff
@@ -3,9 +3,11 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.lightbend.lagom.serialization.Jsonable;
+import lombok.EqualsAndHashCode;
 import lombok.Value;
 
 @Value
+@EqualsAndHashCode(callSuper = false)
 @JsonDeserialize
```

Comment:
```
@Value
@EqualsAndHashCode(callSuper = false)
```

is the combination to avoid producing warnings. The `equals` and `hashCode` produced here will be fine anyway.